### PR TITLE
Config enhacements

### PR DESCRIPTION
**Pull request by MiguelAngelGutierrezMaya**
### Core changes

- Adds optional custom domain support to the Amplify hosting stack by creating an `aws-amplify` `CfnDomain` resource when `AMPLIFY_CUSTOM_DOMAIN` is configured.
- Wires the new domain settings through the v2 infrastructure entrypoint, including support for subdomain mapping via `AMPLIFY_CUSTOM_DOMAIN_PREFIX`.
- Replaces the full-screen loading spinner in the dashboard expense list empty-loading state with skeleton cards, so the UI better matches the final layout while data is loading.
- Introduces a reusable `Skeleton` atom in the shared UI package and a dashboard-specific `ExpenseCardSkeleton` molecule for the expenses flow.

### Schema changes

- None. This PR does not introduce database schema changes, API contract changes, or payload shape changes.

### Minor changes

- Adds unit test coverage for the new Amplify custom domain behavior:
  - no domain resource is created when no custom domain is configured
  - the domain resource is created when a custom domain is provided
  - the configured branch is mapped correctly when a subdomain prefix is used
- Adds unit coverage for the new `Skeleton` and `ExpenseCardSkeleton` components.
- Updates expense list loading-state tests to reflect the skeleton-based behavior.
- Extends `config/.env.defaults` with the new Amplify domain configuration variables.

## Screenshots/Recordings

- No screenshots were attached as part of this change set.
- The infrastructure changes are non-visual.

## How to Test

1. Run the targeted automated checks that cover the scope of this PR:

   ```sh
   pnpm --filter @infra test -- amplify-hosting-stack.test.ts
   pnpm --filter @features/ui test -- skeleton
   pnpm --filter @features/dashboard test -- expense-list expense-card-skeleton
   ```

   Expected result: all test suites pass.

2. Validate the infrastructure configuration path:

   ```sh
   cp config/.env.defaults config/.env.local
   ```

   Update the following values in `config/.env.local`:

   ```sh
   AMPLIFY_CUSTOM_DOMAIN="financial-management.migudev.com"
   AMPLIFY_CUSTOM_DOMAIN_PREFIX="dev"
   ```

   Then synthesize the CDK app:

   ```sh
   cd infra && pnpm cdk synth
   ```

   Expected result: the Amplify hosting stack synthesizes successfully and includes a domain association for `dev.financial-management.migudev.com`.

3. Validate apex-domain behavior by setting the prefix to an empty string:

   ```sh
   AMPLIFY_CUSTOM_DOMAIN_PREFIX=""
   ```

   Re-run:

   ```sh
   cd infra && pnpm cdk synth
   ```

   Expected result: the stack still synthesizes successfully and the branch is mapped directly to the root domain.

4. Validate the dashboard loading experience manually:

   ```sh
   pnpm --filter @client/main web
   ```

   Navigate to the dashboard expenses view and reproduce the state where the list is loading and there are still no expenses in memory.

   Expected result:
   - the previous full-screen spinner is no longer shown in that scenario
   - five expense-card skeletons are rendered instead
   - once data resolves, the skeleton placeholders disappear and the real expense cards render normally
   - if loading finishes with no expenses, the existing empty state is still shown

## Checklist

- [x] The PR description includes clear instructions on how to test the implementation manually.

##### Added Tests?

- [x] yes
  - [x] Verified that this PR is not creating duplicate use cases
  - [x] All edge cases described in the acceptance criteria are covered, and tests have clear and descriptive names
- [ ] no, because they aren't needed
- [ ] no, because I need help

_Override these when needed._

```sh
pnpm --filter @infra test -- amplify-hosting-stack.test.ts
pnpm --filter @features/ui test -- skeleton
pnpm --filter @features/dashboard test -- expense-list expense-card-skeleton
```

### CODE REVIEW

File: config/.env.defaults [hunk 1]
```diff
@@ -72,6 +72,8 @@ ACCESS_TOKEN_NAME="github_token"
 AMPLIFY_ENABLE_AUTO_BUILD="false"
 ASSETS_BUCKET_URL="http://localhost:3000"
 APPLICATION_URL="http://localhost:3000"
+AMPLIFY_CUSTOM_DOMAIN="example.com"
+AMPLIFY_CUSTOM_DOMAIN_PREFIX="dev"
```
[the remark this answers is not in the document]
Comment (MiguelAngelGutierrezMaya) [reply config/.env.defaults:76]: It is not necessary because the process happens on the CI
